pkg/discovery/tracermetadata/tracer_memfd.go

```diff
@@ -0,0 +1,44 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+package tracermetadata
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+)
+
+const memfdTracerFileName = "datadog-tracer-info-"
+const memFdTracerMaxSize = 1 << 16
+
+// HasTracerMetadata returns true if tracer metadata
+// is present for a given process
+func HasTracerMetadata(pid int, procRoot string) bool {
```

Comment:
Is this function used?

Reply:
Doesn't seem needed even for DSCVR-83 since it should ideally check that the metadata is actually valid, and for that purpose you can just use `GetTracerMetadata`.